### ISSUE TITLE
[Bug]: OpenDevin not finding all files in cloned directory

### ISSUE THREAD

**Issue by neubig** [OpenDevin/OpenDevin#3172]
### Is there an existing issue for the same bug?

- [X] I have checked the troubleshooting document at https://docs.all-hands.dev/modules/usage/troubleshooting
- [X] I have checked the existing issues.

### Describe the bug

I'm having a very strange experience where OpenDevin is able to find some but not all files in a directory that was cloned from github. Here is a [full log file](https://www.all-hands.dev/share-opendevin?share_id=aa5b7bb14bb85c079b7d2a023fcf972189c2d2a1b88a52040786f56f4163b2a9&password=YxpXqOX2ZXzzvzAs).

Basically, OpenDevin is able to see some of the subdirectories of the OpenDevin repo that it has cloned, but not all of them. OTOH, when I use `ls` or `cd` directly myself on my OS there is no problem. Any idea what might be going on here?

### Current OpenDevin version

```bash
commit a52df6a27247f09d06176d320c80a7dd28543b35
```

### Installation and Configuration

```bash
make build; make run
```

### Operating System

MacOS: 14.1.2 (23B92)

### Reproduction Steps

See above log for the prompt that I provided.

**Comment by neubig** [comment]
@xingyaoww pointed out that this is because the agent improperly forgot that it cded into the OpenDevin directory, so it was just an agent error.
